tmt/cli.py

```diff
@@ -870,3 +871,129 @@ def status(context, path, abandoned, active, finished, **kwargs):
         raise tmt.utils.GeneralError(f"Path {path} doesn't exist.")
     status_obj = tmt.Status(context=context)
     status_obj.show()
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  Clean
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+dry = tmt.options.force_dry[1]
+
+
+@main.group(chain=True, invoke_without_command=True, cls=CustomGroup)
+@click.pass_context
+@verbose_debug_quiet
+@dry
+def clean(context, **kwargs):
+    """
+    Clean workdirs, guests or images.
+
+    Without any command, clean everything, stop the guests, remove
+    all runs and then remove all images. Search for runs in
+    /var/tmp/tmt, if runs are stored elsewhere, the path to them can
+    be set using a subcommand (either runs or guests subcommand).
+    """
+    clean_obj = tmt.Clean(parent=context.obj, context=context)
+    context.obj.clean = clean_obj
+    exit_code = 0
+    if context.invoked_subcommand is None:
+        echo(style('tmt cleanup', fg='red'))
+        # Set path to default
+        context.params['path'] = tmt.utils.WORKDIR_ROOT
+        # Create another level to the hierarchy so that logging indent is
+        # consistent between the command and subcommands
+        clean_obj = tmt.Clean(parent=clean_obj, context=context)
```

Comment:
As an alternative, instead of creating the artificial hierarchy here, we could possibly introduce a real hierarchy `Clean --> CleanRuns` and use the parent/child relation there. This would also allow calling `CleanRuns.opt('verbose')` and similar. However, I'm not convinced/sure that this rather significant change would be worth the time. Maybe in the future we can polish this?

Reply:
Agreed, this is more of a cosmetic/design change. Lets remember this similarly to `classmethod` and logging issue